ddpui/services/kpi_service.py

```
@@ -0,0 +1,534 @@
+            "target_value": kpi.target_value,
+            "direction": kpi.direction,
+            "green_threshold_pct": kpi.green_threshold_pct,
+            "amber_threshold_pct": kpi.amber_threshold_pct,
+            "time_grain": kpi.time_grain,
+            "time_dimension_column": kpi.time_dimension_column,
+            "trend_periods": kpi.trend_periods,
+        }
+
+        return KPIService.compute_kpi_data(payload, org)
+
+        return periods
```

Ruff (0.15.11)

[error] Undefined name `periods`

(F821)